server/txSender/txSender.go

```diff
@@ -0,0 +1,126 @@
+
+import (
+	"context"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data/sovereign"
+	coreTx "github.com/multiversx/mx-chain-core-go/data/transaction"
+	"github.com/multiversx/mx-sdk-go/core"
+	"github.com/multiversx/mx-sdk-go/data"
+)
+
+// TxSenderArgs holds args to create a new tx sender
+type TxSenderArgs struct {
+	Wallet          core.CryptoComponentsHolder
+	Proxy           Proxy
+	TxInteractor    TxInteractor
+	TxNonceHandler  TxNonceSenderHandler
+	DataFormatter   DataFormatter
+	SCBridgeAddress string
+}
+
+type txSender struct {
+	wallet          core.CryptoComponentsHolder
+	netConfigs      *data.NetworkConfig
+	txInteractor    TxInteractor
+	txNonceHandler  TxNonceSenderHandler
+	dataFormatter   DataFormatter
+	scBridgeAddress string
+}
+
+// NewTxSender creates a new tx sender
+func NewTxSender(args TxSenderArgs) (*txSender, error) {
+	err := checkArgs(args)
+	if err != nil {
+		return nil, err
+	}
+
+	networkConfig, err := args.Proxy.GetNetworkConfig(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	return &txSender{
+		wallet:          args.Wallet,
+		netConfigs:      networkConfig,
+		txInteractor:    args.TxInteractor,
+		dataFormatter:   args.DataFormatter,
+		scBridgeAddress: args.SCBridgeAddress,
+		txNonceHandler:  args.TxNonceHandler,
+	}, nil
+}
+
+func checkArgs(args TxSenderArgs) error {
+	if check.IfNil(args.Wallet) {
+		return errNilWallet
+	}
+	if check.IfNil(args.Proxy) {
+		return errNilProxy
+	}
+	if check.IfNil(args.TxInteractor) {
+		return errNilTxInteractor
+	}
+	if check.IfNil(args.DataFormatter) {
+		return errNilDataFormatter
+	}
+	if check.IfNil(args.TxNonceHandler) {
+		return errNilNonceHandler
+	}
+	if len(args.SCBridgeAddress) == 0 {
+		return errNoSCBridgeAddress
+	}
+
+	return nil
+}
+
+// SendTxs should send bridge data operation txs
+func (ts *txSender) SendTxs(ctx context.Context, data *sovereign.BridgeOperations) ([]string, error) {
+	if len(data.Data) == 0 {
+		return make([]string, 0), nil
+	}
+
+	return ts.createAndSendTxs(ctx, data)
+}
+
+func (ts *txSender) createAndSendTxs(ctx context.Context, data *sovereign.BridgeOperations) ([]string, error) {
+	txHashes := make([]string, 0)
+	txsData := ts.dataFormatter.CreateTxsData(data)
+
+	for _, txData := range txsData {
+		tx := &coreTx.FrontendTransaction{
+			Value:    "1",
+			Receiver: ts.scBridgeAddress,
+			Sender:   ts.wallet.GetBech32(),
+			GasPrice: ts.netConfigs.MinGasPrice,
+			GasLimit: 50_000_000, // todo
+			Data:     txData,
+			ChainID:  ts.netConfigs.ChainID,
+			Version:  ts.netConfigs.MinTransactionVersion,
+		}
+
+		err := ts.txNonceHandler.ApplyNonceAndGasPrice(ctx, ts.wallet.GetAddressHandler(), tx)
+		if err != nil {
+			return nil, err
+		}
+
+		err = ts.txInteractor.ApplyUserSignature(ts.wallet, tx)
+		if err != nil {
+			return nil, err
+		}
+
+		hash, err := ts.txNonceHandler.SendTransaction(ctx, tx)
+		if err != nil {
+			return nil, err
```

Comment:
some transactions might have been sent and error could occur after. Is this case treated anywhere?

Reply:
Should be "treated" by the next leader. If any operation is unsuccessful, next leader will try to bridge them.

Reply:
ok